Mathlib/RingTheory/Noetherian.lean

```diff
@@ -442,14 +442,158 @@ lemma LinearMap.eventually_iSup_ker_pow_eq (f : M →ₗ[R] M) :
   · rw [← hn _ (hm.trans h), hn _ hm]
   · exact f.iterateKer.monotone h.le
 
+namespace Orzech
+
+variable {R M : Type*} [Semiring R] [AddCommMonoid M] [Module R M] {N : Submodule R M}
+  (f : N →ₗ[R] M) (hf : Surjective f) (n : ℕ)
+
+open scoped Classical in
+/-- Zero extension of `f` to `M → M`. -/
+private noncomputable def fext (x : M) : M := if h : x ∈ N then f ⟨x, h⟩ else 0
+
+private theorem fext_zero : fext f 0 = 0 := by
+  rw [fext, dif_pos (zero_mem N)]; exact _root_.map_zero _
+
+/-- Define `D'_n` to be `(f⁻¹ ∘ ⋯ ∘ f⁻¹)(N)` (`n` times`). -/
```

Comment:
Isn't there a more general procedure to iterate something that defining it by induction each time?